it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersChooseAddressControllerISpec.scala

```diff
@@ -1,46 +1,122 @@
-  extends Specifications with TestConfiguration {
+  def route(mode: Mode): String = routes.BusinessPartnersChooseAddressController.load(1, mode).url.drop(6)
 
-  val route = routes.BusinessPartnersChooseAddressController.load().url.drop(6)
+  val index: Int = 1
 
-  s"GET $route" should {
+  private val testAddressData: Map[String, Address] = {
+    val address1 = Address(
+      addressLine1 = "1 Romford Road",
+      addressLine2 = Some("Wellington"),
+      addressLine3 = Some("Telford"),
+      addressLine4 = None,
+      postcode = "TF1 4ER",
+      countryCode = None,
+      lookupId = None
+    )
 
+    val address2 = address1.copy(addressLine1 = "2 Romford Road")
+    val address3 = address1.copy(addressLine1 = "3 Romford Road")
+    val address4 = address1.copy(addressLine1 = "2 Romford Road")
+    val address5 = address1.copy(addressLine1 = "5 Romford Road")
+
+    Map("1" -> address1, "2" -> address2, "3" -> address3, "4" -> address4, "5" -> address5)
+  }
+
+  val seedCacheWithUKAddressLookup: UserAnswers = emptyUserAnswers
+    .set[UkAddressLookup](
+      UkAddressLookupPage(1),
+      UkAddressLookup(
+        Some("1 Romford Road"),
+        "TF1 4ER",
+        testAddressData
+      ))
+    .success
+    .value
+
+  val seedCacheWithSingleUKAddress: UserAnswers = emptyUserAnswers
+    .set[UkAddressLookup](
+      UkAddressLookupPage(1),
+      UkAddressLookup(
+        Some("1 Romford Road"),
+        "TF1 4ER",
+        Map("1" -> Address(
+          addressLine1 = "1 Romford Road",
+          addressLine2 = Some("Wellington"),
+          addressLine3 = Some("Telford"),
+          addressLine4 = None,
+          postcode = "TF1 4ER",
+          countryCode = None,
+          lookupId = None
+        ))
+      ))
+    .success
+    .value
+
+  val seedCacheWithEmptyUKAddressList: UserAnswers = emptyUserAnswers
+    .set[UkAddressLookup](
+      UkAddressLookupPage(1),
+      UkAddressLookup(
+        Some("1 Romford Road"),
+        "TF1 4ER",
+        Map.empty
+      ))
+    .success
+    .value
+
+  // here we need userAnswersWithPageData w the choosen result from user
+
+  List(NormalMode, CheckMode).foreach { mode =>
+    s"GET ${route(mode)}" should {
       "render the choose address page" in {
-        given
-          .commonPrecondition
+        given.commonPrecondition
+        addUserAnswersToSession(seedCacheWithUKAddressLookup)
+
+        val result = buildRequest(route(mode))
+          .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
+          .get()
+
+        whenReady(result) { res =>
+          res.status mustBe 200
+          val page = Jsoup.parse(res.body)
+          page.title() must include("Choose address")
+        }
+      }
 
-        WsTestClient.withClient { client =>
-          val result = client.url(baseUrl + route)
+      "there are no user answers in the database" should {
+        "redirect to the start of BusinessPartners" in {
```

Comment:
reorder to make clearer
e.g.

"redirect to the start of the BusinessPartners" when {
  "there are no user answers in the database" in {
      ...
    }
  }
  
GET ${route(mode)} should redirect to the start of the BusinessPartners when there are no user answers in the database